kartothek/serialization/_parquet.py

```diff
@@ -40,65 +37,9 @@
 def _empty_table_from_schema(parquet_file):
     schema = parquet_file.schema.to_arrow_schema()
 
-    if ARROW_LARGER_EQ_0150:
-        # ARROW-6872: empty dictionary columns raise ArrowNotImplementedError
-        for i in range(len(schema)):
-            field = schema[i]
-            if pa.types.is_dictionary(field.type):
-                new_field = pa.field(
-                    field.name, field.type.value_type, field.nullable, field.metadata
-                )
-                schema = schema.remove(i).insert(i, new_field)
-
     return schema.empty_table()
 
 
-def _reset_dictionary_columns(table, exclude=None):
-    """
-    1. Categorical columns of null won't cast https://issues.apache.org/jira/browse/ARROW-5085
-    2. Massive performance issue due to non-fast path implementation https://issues.apache.org/jira/browse/ARROW-5089
-    3. We need to ensure that the dtype is exactly as requested, see GH227
-    """
-    if exclude is None:
-        exclude = []
-
-    if ARROW_LARGER_EQ_0150:
-        # https://issues.apache.org/jira/browse/ARROW-8142
-        if len(table) == 0:
-            df = table.to_pandas(date_as_object=True)
-            new_types = {
-                col: df[col].cat.categories.dtype
-                for col in df.select_dtypes("category")
-            }
-            if new_types:
-                df = df.astype(new_types)
-                table = pa.Table.from_pandas(df)
-        else:
-            schema = table.schema
-            for i in range(len(schema)):
-                field = schema[i]
-                if pa.types.is_dictionary(field.type):
-                    new_field = pa.field(
-                        field.name,
-                        field.type.value_type,
-                        field.nullable,
-                        field.metadata,
-                    )
-                    schema = schema.remove(i).insert(i, new_field)
-
-            table = table.cast(schema)
-    else:
-        for i in range(table.num_columns):
-            col = table[i]
-            if col.name in exclude:
-                continue
-            if pa.types.is_dictionary(col.type):
-                new_type = col.data.chunk(0).dictionary.type
-                new_col = col.cast(new_type)
-                table = table.remove_column(i).add_column(i, new_col)
-    return table
-
-
```

Comment:
looks like we're hitting https://github.com/JDASoftwareGroup/kartothek/issues/227 again because of this change, see https://github.com/JDASoftwareGroup/kartothek/runs/1025742983

Reply:
Right, I forgot that we sometimes actually want to reset these... what a waste :D